### ISSUE TITLE
Improving performance of the designer

### ISSUE THREAD

**Issue by Bournvita1998** [phpmyadmin/phpmyadmin#15163]
[Designer's JS code related]

I have observed a few performance issues in the designer's JS code like calculating multiples things inside a for loop (which could be calculated outside the loop). Also other than the performance issues, I also observed some bad coding practices in this section, like declaring each variable in a different line(bad coding practice).

**Comment by Bournvita1998** [comment]
Working on it. Will create a PR soon

**Comment by Bournvita1998** [comment]
@williamdes Please review this PR, will work on other files once you approve these changes

**Comment by Bournvita1998** [comment]
@williamdes any reviews?

**Comment by Bournvita1998** [comment]
@williamdes @ibennetch I am planning to create a single PR for the whole refactoring work in the designer section. Good to go?

**Comment by mauriciofauth** [comment]
@Bournvita1998 I don't think it is a good idea, it is very difficult to review such big pull request. It is better to open small pull requests that can be merged into master branch or into a new branch just for this.

**Comment by Bournvita1998** [comment]
@mauriciofauth Would you like to suggest how can that be done?
Can we divide this into different PRs like one for each js file in the designer?

**Comment by mauriciofauth** [comment]
@Bournvita1998 I created the [`gsoc-db-designer`](https://github.com/phpmyadmin/phpmyadmin/tree/gsoc-db-designer) branch for now, so you can open PRs against this branch.

cc @ibennetch
